build fails on VS2012 and earlier

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master ( 3cee54cd2 )
- Operating System / Platform => Windows 7 Professional 64bit 
- Compiler => Visual Studio 2012

##### Detailed description

 - I tried to build recent master, and faced to following error
```
D:\Jenkins\OpenCV-MSBuild\workspace\modules\calib3d\src\ap3p.cpp(35): error C3861: 'cbrt': identifier not found 
```
 - I figured out that ```cbrt``` function is only added on [Visual Studio 2013 and after](https://msdn.microsoft.com/ja-jp/library/dn465177(v=vs.120).aspx)
 - I also found ```cv::cubeRoot``` so I think in old compiler, this would be an alternative
 - I'll create a PR later

##### Steps to reproduce
 - ```cmake``` and ```make``` as usual

--- a/modules/calib3d/src/ap3p.cpp
+++ b/modules/calib3d/src/ap3p.cpp
@@ -2,6 +2,9 @@
 
 #include <cmath>
 #include <complex>
+#if defined (_MSC_VER) && (_MSC_VER <= 1700)
+static inline double cbrt(double x) { return (double)cv::cubeRoot((float)x); };
+#endif
 
 using namespace std;
 

